src/directives/formly-form.js

```diff
@@ -188,6 +188,9 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
             field);
         }
       }
+      if (field.model) {
+        $scope.$watch(() => field.model, onModelOrFormStateChange, true);
```

Comment:
Ah, very good. Sounds like a reasonable optimization to me. You'd probably need to do a separate `angular.forEach` above in that case. If you decide to do that, please write a test case to cover that :-)